### PR TITLE
feat(storcon): forward gc blocking and unblocking

### PR DESCRIPTION
**Pull request by koivunej**
Currently using gc blocking and unblocking with storage controller managed pageservers is painful. Implement the API on storage controller.

Fixes: #8893

### CODE REVIEW

File: test_runner/regress/test_timeline_gc_blocking.py [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
 from fixtures.pageserver.utils import wait_timeline_detail_404
 
 
-def test_gc_blocking_by_timeline(neon_env_builder: NeonEnvBuilder):
+@pytest.mark.parametrize("sharded", [True, False])
+def test_gc_blocking_by_timeline(neon_env_builder: NeonEnvBuilder, sharded: bool):
+    neon_env_builder.num_pageservers = 2 if sharded else 1
     env = neon_env_builder.init_start(
-        initial_tenant_conf={"gc_period": "1s", "lsn_lease_length": "0s"}
+        initial_tenant_conf={"gc_period": "1s", "lsn_lease_length": "0s"},
```
Comment (arpad-m) [inline test_runner/regress/test_timeline_gc_blocking.py:20]: why are you setting `lsn_lease_length` here?

File: test_runner/regress/test_timeline_gc_blocking.py [hunk 2]
```diff
@@ -34,34 +48,79 @@ def test_gc_blocking_by_timeline(neon_env_builder: NeonEnvBuilder):
...
+        old = self.offset
+        self.offset = offset
+        log.info(f"{old} -> {offset}: {msg}")
+
+    def assert_log_does_not_contain(self, what: str):
+        assert self.pageserver.log_contains(what) is None
+
+
+@dataclass(frozen=True)
+class ManyPageservers:
```
Comment (arpad-m) [inline test_runner/regress/test_timeline_gc_blocking.py:100]: This should (together with `ScrollableLog`) maybe be somewhere more reusable

**Review by arpad-m — APPROVED** [review]

File: test_runner/regress/test_timeline_gc_blocking.py (hunk 1 above)
Comment (koivunej) [reply test_runner/regress/test_timeline_gc_blocking.py:20]: initial wait time is random sleep up to gc interval and the lsn lease length.

this basically makes the test start up "immediatedly" (heavy sleeps though).

File: test_runner/regress/test_timeline_gc_blocking.py (hunk 2 above)
Comment (koivunej) [reply test_runner/regress/test_timeline_gc_blocking.py:100]: we already have so much crap in the fixtures, maybe follow the rule of 3 with generalizing this. this is the first tedt I've written in this style.

**Pull request merged**